File: 2_array/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // prompt the user for text input
    string text = get_string("Text: ");

    // count number of letters, words and sentences
    int letters = 0;
    int words = 0;
    int sentences = 0;
    int spaces = 0;
    int puncts = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isspace(text[i]))
        {
            spaces++;
        }
        else if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            puncts++;
        }
        else
        {
            continue;
        }
    }
    words = spaces + 1;
    sentences = puncts;

    /*
    printf("%i letter(s)\n", letters);
    printf("%i word(s)\n", words);
    printf("%i sentence(s)\n", sentences);
    */

    // calculate index
    float L = ((float) letters * 100) / (float) words;
    float S = ((float) sentences * 100) / (float) words;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    /*
    printf("%f\n", L);
    printf("%f\n", S);
    printf("%f\n", index);
    */

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }
}

File: 6_python/pset6/readability/readability.py
from cs50 import get_string

# prompt user for input
text = get_string("Text: ")

# init variables
letters = 0
spaces = 0
puncts = 0

# iter through text
for i in text:
    if i.isalpha() == True:
        letters += 1
    elif i.isspace() == True:
        spaces += 1
    elif i in [".", "!", "?"]:
        puncts += 1
    else:
        continue

# other variables
words = spaces + 1
sentences = puncts

# calculate index
L = (letters * 100) / words
S = (sentences * 100) / words
index = 0.0588 * L - 0.296 * S - 15.8

# output according to index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")